test/test_enrolments_controller.py

```diff
@@ -0,0 +1,123 @@
+from test.party_client import PartyTestClient
+
+from sqlalchemy.exc import DataError
+from sqlalchemy.orm.exc import NoResultFound
+
+from ras_party.controllers.enrolments_controller import respondent_enrolments
+from ras_party.models.models import (
+    Business,
+    BusinessRespondent,
+    Enrolment,
+    EnrolmentStatus,
+    Respondent,
+)
+from ras_party.support.session_decorator import with_db_session
+
+respondents_enrolments = [
+    {
+        "respondent": "b6f9d6e8-b840-4c95-a6ce-9ef145dd1f85",
+        "enrolment_details": [
+            {
+                "business": "75d9af56-1225-4d43-b41d-1199f5f89daa",
+                "survey_id": "9200d295-9d6e-41fe-b541-747ae67a279f",
+                "status": EnrolmentStatus.ENABLED,
+            },
+            {
+                "business": "98e2c9dd-a760-47dd-ba18-439fd5fb93a3",
+                "survey_id": "c641f6ad-a5eb-4d82-a647-7cd586549bbc",
+                "status": EnrolmentStatus.ENABLED,
+            },
+        ],
+    },
+    {
+        "respondent": "5718649e-30bf-4c25-a2c0-aaa733e54ed6",
+        "enrolment_details": [
+            {
+                "business": "af25c9d5-6893-4342-9d24-4b88509e965f",
+                "survey_id": "9200d295-9d6e-41fe-b541-747ae67a279f",
+                "status": EnrolmentStatus.ENABLED,
+            },
+            {
+                "business": "75d9af56-1225-4d43-b41d-1199f5f89daa",
+                "survey_id": "9200d295-9d6e-41fe-b541-747ae67a279f",
+                "status": EnrolmentStatus.DISABLED,
+            },
+        ],
+    },
+]
+
+
+class TestEnrolments(PartyTestClient):
+
+    def setUp(self):
+        self._add_enrolments()
+
+    def test_get_enrolments_party_id(self):
+        enrolments = respondent_enrolments(party_uuid="b6f9d6e8-b840-4c95-a6ce-9ef145dd1f85")
+
+        self.assertEqual(len(enrolments), 2)
+        self.assertIn(str(enrolments[0].business_id), "75d9af56-1225-4d43-b41d-1199f5f89daa")
+        self.assertIn(str(enrolments[1].business_id), "98e2c9dd-a760-47dd-ba18-439fd5fb93a3")
+
+    def test_get_enrolments_party_id_and_business_id_and_survey_id(self):
+        enrolments = respondent_enrolments(
+            party_uuid="b6f9d6e8-b840-4c95-a6ce-9ef145dd1f85",
+            business_id="75d9af56-1225-4d43-b41d-1199f5f89daa",
+            survey_id="9200d295-9d6e-41fe-b541-747ae67a279f",
+        )
+
+        self.assertEqual(len(enrolments), 1)
+        self.assertIn(str(enrolments[0].respondent_id), "b6f9d6e8-b840-4c95-a6ce-9ef145dd1f85")
```

Comment:
This assertion is inadvertently passing. The `str(enrolments[0].respondent_id)` correctly yields the integer `1` (i.e. the db sequence key for the respondent, but we're then asserting that the string value `1` is in the UUID string `b6f9d6e8-b840-4c95-a6ce-9ef145dd1f85` which it is.

These assertions could all be `assertEqual` with the respondent_id being changed to
```
self.assertIn(str(enrolments[0].respondent_id), "1")
```

Reply:
That is a cracking spot, and you're right it should be assertEqual for everything. I must of just copied and pasted after initially getting it wrong or doing it differently